checkov/terraform/utils.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+from typing import Optional
+
+
+class TFModule:
+    __slots__ = ("path", "name", "foreach_idx")
+
+    def __init__(self, path: str, name: str, foreach_idx: Optional[int | str] = None) -> None:
+        self.path = path
+        self.name = name
+        self.foreach_idx = foreach_idx
+
+    def __eq__(self, other) -> bool:
```

Comment:
```suggestion
    def __eq__(self, other: Any) -> bool:
        if not isinstance(other, TFModule):
            return False
```
same for the other two 🙂 better to be safe than sorry 😄